### ISSUE TITLE
env stanza at group and job level

### ISSUE THREAD

**Issue by pznamensky** [hashicorp/nomad#10812]
### Use-cases
Execute lots of tasks with set of common environment variables.

For instance, we had several containers (or sometimes groups) in a job and they have common variables.
Right now we had to copy and paste them in env stanza in each task level block.
It leads to errors when we want to change the same variable for each task in a job and maintaining such a big nomad file is not very convenient.

A good example is Airflow.
Take a look at this docker-compose file, where the problem with reusing block solved quite elegantly using YAML power, and try to imagine how wordy and not readable it will look in a nomad file:
https://airflow.apache.org/docs/apache-airflow/stable/docker-compose.yaml

### Attempted Solutions
Right now the most copy-paste block in our job files is `env`.
So it would be really helpful to be able to share a common environment at least at a job and group level.
In other words, it would be great to allow `env` stanza at group and job level too.
